modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/StepItems.scala

```diff
@@ -38,45 +37,40 @@ object StepItems {
   val gpiDispersers: Map[String, String] =
     GpiDisperser.all.map(x => x.shortName -> x.longName).toMap
 
-  private val obsNames =
+  private val obsNames                   =
     GpiObservingMode.all.map(x => x.shortName -> x.longName).toMap
 
   implicit class StepOps(val s: Step) extends AnyVal {
-    def fpuNameMapper(i: Instrument): String => Option[String] = i match {
-      case Instrument.GmosS => enumerations.fpu.GmosSFPU.get
-      case Instrument.GmosN => enumerations.fpu.GmosNFPU.get
-      case Instrument.F2    => enumerations.fpu.Flamingos2.get
-      case _                => _ => none
-    }
+    def fpuNameMapper(i: Instrument): String => Option[String] =
+      i match {
+        case Instrument.GmosS => enumerations.fpu.GmosSFPU.get
+        case Instrument.GmosN => enumerations.fpu.GmosNFPU.get
+        case Instrument.F2    => enumerations.fpu.Flamingos2.get
+        case _                => _ => none
+      }
 
     def exposureTime: Option[Double] = observeExposureTimeO.getOption(s)
+
     def exposureTimeS(i: Instrument): Option[String] =
       exposureTime.map(formatExposureTime(i))
+
     def exposureAndCoaddsS(i: Instrument): Option[String] =
       (coAdds, exposureTime) match {
         case (c, Some(e)) if c.exists(_ > 1) =>
-          s"${c.foldMap(_.show)}x${formatExposureTime(i)(e)} [s]".some
+          s"${c.foldMap(_.show)}x${formatExposureTime(i)(e)} s".some
         case (_, Some(e))                    =>
-          s"${formatExposureTime(i)(e)} [s]".some
+          s"${formatExposureTime(i)(e)} s".some
         case _                               => none
       }
     def coAdds: Option[Int] = observeCoaddsO.getOption(s)
 
-    def isNoneFPU(i: Instrument): Boolean =
-      (i, instrumentFPUO.getOption(s)) match {
-        case (Instrument.GmosS | Instrument.GmosN | Instrument.F2, Some("FPU_NONE")) => true
-        case _ => false
-      }
-
     def fpu(i: Instrument): Option[String] =
-      for {
-        mode <- instrumentFPUModeO
-          .getOption(s)
-          .orElse(FPUMode.BuiltIn.some) // If the instrument has no fpu mode default to built in
-        fpuL = if (isNoneFPU(i) || mode === FPUMode.BuiltIn) instrumentFPUO
-        else instrumentFPUCustomMaskO
-        fpu <- fpuL.getOption(s)
-      } yield fpuNameMapper(i)(fpu).getOrElse(fpu)
+      (i, instrumentFPUO.getOption(s), instrumentFPUCustomMaskO.getOption(s)) match {
+        case (Instrument.GmosS | Instrument.GmosN | Instrument.F2, Some("CUSTOM_MASK"), c) => c
+        case (Instrument.GmosS | Instrument.GmosN | Instrument.F2, None, c @ Some(_))      => c
+        case (_, Some(b), _)                                                               => fpuNameMapper(i)(b)
+        case _                                                                             => none
+      }
```

Comment:
If I understand correctly, we are not defaulting to `BuiltIn` anymore, is that the fix?

Reply:
There are several options how the OT presents the sequence depending on whether there are iterators and what not.
This basically tries to find if there is a custom mask present first and if not it goes the build in
There is a flag to indicate whether we use custom or built in and it was used previously
but it turns out it is unreliable and there are cases where the flag indicates we have a built in mask
while in reality it is custom